dGame/dComponents/RebuildComponent.cpp

```diff
@@ -198,9 +198,9 @@ void RebuildComponent::Update(float deltaTime) {
 			if (!destComp) break;
 
 			++m_DrainedImagination;
-			const int32_t imaginationCostRemaining = m_TakeImagination - m_DrainedImagination;
+			const auto imaginationCostRemaining = m_TakeImagination - m_DrainedImagination;
 
-			const int32_t newImagination = destComp->GetImagination() - 1;
+			const auto newImagination = destComp->GetImagination() - 1;
```

Comment:
I dont think we should change these, they work currently and should be forced ints on the off chance both the operands are unsigned and the subtraction underflows.